NBitcoin/Protocol/NodeSocketEventManager.cs

```diff
@@ -0,0 +1,62 @@
+﻿using System;
+using System.Net;
+using System.Net.Sockets;
+using System.Threading;
+
+namespace NBitcoin.Protocol
+{
+	/// <summary>
+	/// Singleton class which deals with a single instance of <see cref="SocketAsyncEventArgs"/>.
+	/// </summary>
+	internal sealed class NodeSocketEventManager : IDisposable
+	{
+		private SocketAsyncEventArgs instance;
+
+		private NodeSocketEventManager() { }
+
+		/// <summary>
+		/// Creates an instance of <see cref="SocketAsyncEventArgs"/>.
```

Comment:
Not really, you're creating an instance of NodeSocketEventManager that, as it happens, contains an instance of SocketAsyncEventArgs.

Reply:
Fixed